modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AbstractAccessControlProvider.java

```diff
@@ -86,16 +86,15 @@ protected void checkForCyclicDependencies(AccessControlGroup group, Set<AccessCo
    */
   protected void collectAccessControls(AccessControlGroup group, Set<AccessControlGroup> toplevelGroups) {
 
-    // TODO hohwille Java HashMap buggy???
```

Comment:
As heavily discussed in #241 we have a general problem with `equals()` and `hashCode()`. Here the access controls are added to a `Set` and in case of groups they are modified after that what changes the hashCode. The result is that after the change the original object can neither be found nor removed from the `Set`. Therefore the code is IMHO now even more buggy and might cause the exception in situations where it is undesired.
As the problem is going beyond the actual `if` and the `TODO` I created a new issue: #418.
Unless #418 is fixed, I would stay with the `if` statement being inactive to avoid nasty trouble.